protocol/xmrmaker/errors.go

```diff
@@ -28,11 +29,55 @@ var (
 
 	// protocol initiation errors
 	errProtocolAlreadyInProgress = errors.New("protocol already in progress")
-	errBalanceTooLow             = errors.New("balance lower than amount to be provided")
 	errNoOfferWithID             = errors.New("failed to find offer with given ID")
 	errOfferIDNotSet             = errors.New("offer ID was not set")
-	errAmountProvidedTooLow      = errors.New("amount provided by taker is too low for offer")
-	errAmountProvidedTooHigh     = errors.New("amount provided by taker is too high for offer")
-	errUnlockedBalanceTooLow     = errors.New("unlocked balance is less than maximum offer amount")
 	errSwapCompleted             = errors.New("swap is already completed")
 )
+
+type errBalanceTooLow struct {
+	unlockedBalance float64
+	providedAmount  float64
+}
+
+func (e errBalanceTooLow) Error() string {
+	return fmt.Sprintf("balance of %s XMR is below provided %s XMR",
+		strconv.FormatFloat(e.unlockedBalance, 'f', -1, 64),
+		strconv.FormatFloat(e.providedAmount, 'f', -1, 64),
```

Comment:
The `FormatFloat` call above eliminates trailing zeros after the decimal point which, apparently, there is no good way to do using `%` options to `Sprintf`.  In some places we are using `%v`, but that is not the same and can sometimes turn into exponent notation.  When we create a type for the float64 representation of XMR, maybe we can add a `String()` function so we don't have to repeat this code.